lib/spack/spack/solver/asp.py

```diff
@@ -4331,13 +4268,10 @@ def _is_reusable(spec: spack.spec.Spec, packages, local: bool) -> bool:
 
 
 def _has_runtime_dependencies(spec: spack.spec.Spec) -> bool:
-    if not WITH_RUNTIME:
-        return True
-
-    if spec.compiler.name == "gcc" and not spec.dependencies("gcc-runtime"):
+    if "gcc" in spec and "gcc-runtime" not in spec:
         return False
 
-    if spec.compiler.name == "oneapi" and not spec.dependencies("intel-oneapi-runtime"):
+    if "intel-oneapi-compilers" in spec and "intel-oneapi-runtime" not in spec:
```

Comment:
We should make a note to make this less specific -- I think this is only called to determine reusability so maybe a case for a package property (maybe we need, e.g. `reusable` and `dependents_reusable`?) shortly after this PR.